src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/scheduler/BlobProcessor.java

```diff
@@ -0,0 +1,143 @@
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.DocumentManagementService;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.output.Pdf;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.EntityParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+
+/**
+ * This class is a scheduler which runs as per configured interval.
+ * It will read all the blobs from Azure Blob storage and will do below things
+ * 1. Reads Blob from container.
+ * 2. Extract Zip file(Blob)
+ * 3. Transform metadata json to DB entities.
+ * 4. Save PDF files in document storage.
+ * 5. Update status and doc urls in DB.
+ */
+@Component
+@ConditionalOnProperty(value = "scheduling.enabled", matchIfMissing = true)
+public class BlobProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(BlobProcessor.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final EnvelopeRepository envelopeRepository;
+    private final DocumentManagementService documentManagementService;
+
+    @Autowired
+    public BlobProcessor(
+        CloudBlobClient cloudBlobClient,
+        EnvelopeRepository envelopeRepository,
+        DocumentManagementService documentManagementService
+    ) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.envelopeRepository = envelopeRepository;
+        this.documentManagementService = documentManagementService;
+    }
+
+    @Scheduled(fixedDelayString = "${scan.delay}")
+    public void processBlobs() {
+        cloudBlobClient.listContainers()
+            .forEach(cloudBlobContainer ->
+                processBlobsPerContainer(cloudBlobContainer.getName())
+            );
+    }
+
+    private void processBlobsPerContainer(String containerName) {
+        log.info("Processing blobs for container {} ", containerName);
+
+        try {
+            processZipFiles(cloudBlobClient.getContainerReference(containerName));
+        } catch (URISyntaxException | StorageException e) {
+            log.error("Exception occurred while retrieving container reference for " + containerName, e);
+            // TODO Raise an event in DB or track in appinsights if this happens.
+        }
+    }
+
+    private void processZipFiles(CloudBlobContainer container) {
+        for (ListBlobItem blobItem : container.listBlobs()) {
+            List<Pdf> pdfFiles = new ArrayList<>();
+            String zipFilename = FilenameUtils.getName(blobItem.getUri().toString());
+            byte[] metadataStream = null;
+
+            try {
+                BlobInputStream blobInputStream =
+                    container.getBlockBlobReference(zipFilename).openInputStream();
```

Comment:
Looks like this input stream needs to be in `try with resource` block. Or is it closed all together with `ZipInputStream` couple lines below?